### ISSUE TITLE
Rewrite in rust?

### ISSUE THREAD

**Issue by shadowcat-mst** [trg404/BitList#4]
English is vague, under specified, and generally hard to understand - this could easily lead to problems up to and including the audience being unintentionally offended or worse still failing to get the joke.

With rust's better specification and capacity for type punning, borrow checker to mark jokes taken from other comedians so you don't accidentally both reference them during the same set, and explicit memory model so you can't forget a joke half way through, it would clearly be a far superior approach.

Additionally, cargo would be a vast improvement over your usual choice of pants.

So, please, rewrite it in rust! (ref: https://transitiontech.ca/random/RIIR )

**Comment by genehack** [comment]
Instead of Rust you should consider [Lojban](https://en.wikipedia.org/wiki/Lojban).

**Comment by 64** [comment]
Yes, I strongly agree. English isn't memory-safe. With rust, you'll never forget your routine again.

**Comment by Kixunil** [comment]
In addition, Rust is fearlessly concurrent. English is not - just try telling two jokes at the same time and you'll see how difficult is to do it without data races.

**Comment by luciusmagn** [comment]
With Rust's amazing ownership system you will never have to worry about your jokes being stolen again!
